llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -1477,6 +1479,34 @@ void VFABI::getVectorVariantNames(
   }
 }
 
+std::optional<std::pair<FunctionType *, int>>
+VFABI::createFunctionType(const VFInfo &Info, const FunctionType *ScalarFTy,
+                          Type *VecRetTy, const Module *M) {
```

Comment:
Shouldn't the return type of the vector function also be constructed from `ScalarFTy`?

Reply:
Discussed in person and decided to drop `VecRetTy`.

`VFInfo` does not hold such information since mangled names do not encode it,
however, if that changes then it will. So one should be able to get it from there.